tests/unit/test_endpoint_provider.py

```diff
@@ -272,9 +277,7 @@ def test_a_bad_alert_expression_logs_an_error(self, _):
             self.harness.add_relation_unit(rel_id, "provider/0")
             messages = sorted(logger.output)
             self.assertEqual(len(messages), 1)
-            self.assertIn(
-                "Invalid alert rule missing_expr.rule: missing an 'expr' property", messages[0]
-            )
+            self.assertIn("Invalid rules file: missing_expr.rule", messages[0])
```

Comment:
There is a loss of detail here as to why it failed. Why?

Reply:
What I mean is that the new message is worse. The previous one actually revealed what the user needs to do to fix the error.

Reply:
The message changed because validity is more complex now, per `_is_custom_alert_rule_format()`.
Format is referenced in the module level docstring which is rendered on charmhub.
When `_is_custom_alert_rule_format()` checks better for validity (e.g. pydantic against a schema) then it would be easier to provide the precise reason.